code/modules/power/singularity/emitter.dm

```diff
@@ -256,90 +273,92 @@
 
 	return ..()
 
-/obj/machinery/power/emitter/wrench_act(mob/living/user, obj/item/I)
-	..()
-	default_unfasten_wrench(user, I)
+/obj/machinery/power/emitter/wrench_act(mob/living/user, obj/item/item)
+	. = ..()
+	default_unfasten_wrench(user, item)
 	return TRUE
 
-/obj/machinery/power/emitter/welder_act(mob/living/user, obj/item/I)
-	..()
+/obj/machinery/power/emitter/welder_act(mob/living/user, obj/item/item)
+	. = ..()
 	if(active)
 		to_chat(user, "<span class='warning'>Turn [src] off first!</span>")
 		return TRUE
 
 	if(welded)
-		if(!I.tool_start_check(user, amount=0))
+		if(!item.tool_start_check(user, amount=0))
 			return TRUE
 		user.visible_message("<span class='notice'>[user.name] starts to cut the [name] free from the floor.</span>", \
 			"<span class='notice'>You start to cut [src] free from the floor...</span>", \
 			"<span class='hear'>You hear welding.</span>")
-		if(I.use_tool(src, user, 20, volume=50) && welded)
-			welded = FALSE
-			to_chat(user, "<span class='notice'>You cut [src] free from the floor.</span>")
-			disconnect_from_network()
-			update_cable_icons_on_turf(get_turf(src))
-
-	else if(anchored)
-		if(!I.tool_start_check(user, amount=0))
-			return TRUE
-		user.visible_message("<span class='notice'>[user.name] starts to weld the [name] to the floor.</span>", \
-			"<span class='notice'>You start to weld [src] to the floor...</span>", \
-			"<span class='hear'>You hear welding.</span>")
-		if(I.use_tool(src, user, 20, volume=50) && anchored)
-			welded = TRUE
-			to_chat(user, "<span class='notice'>You weld [src] to the floor.</span>")
-			connect_to_network()
-			update_cable_icons_on_turf(get_turf(src))
+		if(!item.use_tool(src, user, 20, volume=50) || !welded)
+			return
+		welded = FALSE
+		to_chat(user, "<span class='notice'>You cut [src] free from the floor.</span>")
+		disconnect_from_network()
+		update_cable_icons_on_turf(get_turf(src))
+		return TRUE
 
-	else
+	if(!anchored)
 		to_chat(user, "<span class='warning'>[src] needs to be wrenched to the floor!</span>")
-
+		return TRUE
+	if(!item.tool_start_check(user, amount=0))
+		return TRUE
+	user.visible_message("<span class='notice'>[user.name] starts to weld the [name] to the floor.</span>", \
+		"<span class='notice'>You start to weld [src] to the floor...</span>", \
+		"<span class='hear'>You hear welding.</span>")
+	if(!item.use_tool(src, user, 20, volume=50) || !anchored)
```

Comment:
is 20 in this case a unit of time?

Reply:
no, is the amount, i'll clarify it by doing amount=20